### DISCUSSION TITLE
set_to_none=True and accumulate_grad_batches

### DISCUSSION THREAD

**Discussion by denix56** [Lightning-AI/pytorch-lightning#6703] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Is it possible to use automatic optimization with accumulate_grad_batches and performance trick zero_grad(set_to_none=True)?

**Comment by awaelchli** [comment, marked as answer]
Yes, you can override the zero_grad hook in LightningModule

```python
    def optimizer_zero_grad(self, epoch: int, batch_idx: int, optimizer: Optimizer, optimizer_idx: int):
        optimizer.zero_grad(set_to_None=True)
```

**Reply by akihironitta** [reply]
https://pytorch-lightning.readthedocs.io/en/latest/benchmarking/performance.html#zero-grad-set-to-none-true
